### PR TITLE
docs: Added new recommendations for Azure machine learning

### PR DESCRIPTION
**Pull request by sonmitt**
# Overview/Summary

Added new recommendations for Azure Machine Learning.

## Related Issues/Work Items

AB#[38370](https://dev.azure.com/CSUSolEng/Accelerator%20-%20AI/_workitems/edit/38370/)

### Breaking Changes

1. Added new recommendations for ML workspace and associated resources
2. Updated index.md file with the link to dependencies to Container registry

## As part of this pull request I have

- [X] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [X] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [X] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: azure-resources/MachineLearningServices/workspaces/_index.md [hunk 1]
```diff
@@ -1,7 +1,17 @@
 ---
 title: workspaces
 geekdocCollapseSection: true
-geekdocHidden: true
+geekdocHidden: false
 ---
 
+## Dependent Azure Resource Recommendations
```
[the remark this answers is not in the document]
Comment (sonmitt) [reply azure-resources/MachineLearningServices/workspaces/_index.md:7]: Thanks @ejhenry for review. I have a follow-up question.
Azure Machine Learning workspace relies on several dependent resources, such as Azure Storage Account, Key Vault, App Insights, Compute, and Container Registry. Some of the common resiliency recommendations apply to these resources as well. Should we repeat these recommendations for the ML Workspace if we can't link to the existing ones?

[the remark this answers is not in the document]
Comment (ejhenry) [reply azure-resources/MachineLearningServices/workspaces/_index.md:7]: I'm not sure if I follow your question. If the recommendations for those other services already exist in APRL, then the resources would just need to be included in the scope used for the collector script.

**Review by ejhenry — APPROVED** [review]
thanks @sonmitt LGTM

**Review by oZakari — APPROVED** [review]
LGTM

**Pull request merged**